File: selenium/tests/login.spec.rb
require_relative '../env'
require_relative '../pages/login.page'

RSpec.describe 'Login' do

  before(:example) do
    $driver.get 'https://www.saucedemo.com/'
  end

  it 'Login com sucesso' do
    login('standard_user', 'secret_sauce')
    $driver.find_element(:id, 'inventory_filter_container')
  end

  it 'Login com usuário e/ou senha incorretos' do
    login('standard_user', '123456')
    $driver.find_element(:class, 'error-button')
  end

end

File: cypress/cypress/integration/cypress_poc/login.spec.js

import { login } from "./pages/login.page"

describe('Login', () => {

  beforeEach(() => {
    cy.visit('https://www.saucedemo.com/')
  })

  it('Login com sucesso', () => {
    login('standard_user', 'secret_sauce')
    cy.get('#inventory_filter_container')
  })

  it('Login com usuário e/ou senha incorretos', () => {
    login('standard_user', '123456')
    cy.get('.error-button')
  })

})
